bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/scanner/RowResultScanner.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2019 Google LLC. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.bigtable.grpc.scanner;
+
+import com.google.api.gax.rpc.ServerStream;
+import com.google.cloud.bigtable.metrics.BigtableClientMetrics;
+import com.google.cloud.bigtable.metrics.Meter;
+import com.google.cloud.bigtable.metrics.Timer;
+import java.util.ArrayList;
+import java.util.Iterator;
+
+/**
+ * A {@link ResultScanner} that wraps GCJ {@link ServerStream}.
+ */
+public class RowResultScanner<T> implements ResultScanner<T> {
+
+  private static final Meter resultsMeter =
+      BigtableClientMetrics.meter(BigtableClientMetrics.MetricLevel.Info, "scanner.results");
+  private static final Timer resultsTimer = BigtableClientMetrics
+      .timer(BigtableClientMetrics.MetricLevel.Debug, "scanner.results.latency");
+
+  private final ServerStream<T> stream;
+  private final Iterator<T> iterator;
+  private final T[] arr;
+
+  /**
+   * @param stream a Stream of type Row/FlatRow.
+   * @param arr An array of type T length zero.
+   */
+  public RowResultScanner(ServerStream<T> stream, T[] arr) {
+    this.stream = stream;
+    this.iterator = stream.iterator();
+    this.arr = arr;
+  }
+
+  @Override
+  public T next() {
+    if (!iterator.hasNext()) {
+      return null;
+    }
+
+    try (Timer.Context ignored = resultsTimer.time()) {
+      T result = iterator.next();
+      resultsMeter.mark();
+      return result;
+    } catch (RuntimeException e) {
+      throw e;
+    }
+  }
+
+  @Override
+  public T[] next(int count) {
+    ArrayList<T> resultList = new ArrayList<>(count);
+    for (int i = 0; iterator.hasNext() && i < count; i++) {
+      T row = next();
+      if (row == null) {
+        break;
+      }
+      resultList.add(row);
+    }
+    return resultList.toArray(arr);
+  }
+
+  @Override
+  public int available() {
+    throw new UnsupportedOperationException("this operation is not supported");
```

Comment:
@igorbernstein2, is there anyway to get the in-memory count of available rows?

Reply:
No, the  gcj client only parses rows on demand. Closest you can do is `iterator.hasNext() ? 1 : 0;`
Remind me how this is used?

Reply:
`ResultScanner#availble` returns the number of rows available without blocking the network. 
Should we use `stream.isReceiveReady() ? 1 : 0` ? as this will be a non-blocking call.
@sduskis 
Please let me know if we should return a value or continue to throw an exception.

Reply:
Please return a value.

Reply:
Have updated this method with `stream.isReceiveReady() ? 1 : 0`